docroot/modules/custom/civictheme_migrate/src/CivicthemeMigrateManager.php

```diff
@@ -133,24 +133,30 @@ public function generateMigration(array $file_ids) {
     foreach ($files as $file) {
       $file_stream_wrappers[] = $file->getFileUri();
     }
-    $migration_config_file_name = $this->extensionPathResolver->getPath('module', 'civictheme_migrate') . '/assets/migrations/migrate_plus.migration.civictheme_page.yml';
-    $migration_config = file_get_contents($migration_config_file_name);
-    $migration_config = $this->yaml->decode($migration_config);
-    $migration_config['source']['urls'] = $file_stream_wrappers;
-    $existing_migration = $this->migrationStorage->load('civictheme_page_migrate');
-    if ($existing_migration === NULL) {
-      $migration = $this->migrationStorage->create($migration_config);
-      $migration->save();
-      $this->messenger()->addStatus($this->t('Migration has been generated.'));
-    }
-    else {
-      // Adding all keys in case we update the asset migration template.
-      $existing_migration->set('process', $migration_config['process']);
-      $existing_migration->set('source', $migration_config['source']);
-      $existing_migration->set('destination', $migration_config['destination']);
-      $existing_migration->save();
-      $this->messenger()->addStatus($this->t('Migration has been updated.'));
+    $migration_directory = $this->extensionPathResolver->getPath('module', 'civictheme_migrate') . '/assets/migrations/';
+    $migration_config_files = glob($migration_directory . '*.yml');
+    foreach ($migration_config_files as $migration_config_file) {
```

Comment:
Expanded to handle multiple migrations - we are completing media migrations using the same JSON files as the content - I have noticed on other merlin migrations they have separate media extracted content JSON - not sure if there is a critical reason for this